backend/inspirehep/records/marshmallow/literature/bibtex.py

```diff
@@ -246,9 +248,23 @@ def get_series(self, data):
         return get_value(data, "book_series.title[0]")
 
     def get_book_title(self, data):
-        book_series_title = get_value(data, "book_series.title[0]")
-        conference_record_title = get_value(data, "titles.title[0]")
-        return book_series_title or conference_record_title
+        try:
+            book_records = InspireRecord.get_linked_records_from_dict_field(
+                data, "publication_info.parent_record"
+            )
+            book_record = next(book_records, None)
+            return get_value(book_record, "titles.title[0]")
+        except TypeError:
```

Comment:
Is this to guard against `book_record` being `None`? You can fix this by doing `book_record = next(book_record, {})` instead.

Reply:
Also `get_linked_records_from_dict_field` returns an empty list when there's no linked record, which throws an error when calling `next` on it. With the try-except I can also avoid checking if it returns a generator or an empty list.